app/controllers/autocomplete_controller.rb

```diff
@@ -94,7 +94,7 @@ def noncanonical_tag
     tag_class = params[:type].classify.constantize
     one_tag = tag_class.find_by(canonical: false, name: params[:term])
     # If there is an exact match in the database, ensure it is the first thing suggested.
-    match = if one_tag
+    match = if one_tag && params[:term].present?
```

Comment:
Just one question! (This time, from the right account.) Why does this need to be here and not on line 95, like

```
one_tag = tag_class.find_by(canonical: false, name: params[:term]) if params[:term].present?
```

Reply:
Either spot should work, but your suggestion is likely slightly better because it skips the database query. Changed!